libraries/db/include/graphene/db/undo_database.hpp

```diff
@@ -66,7 +66,7 @@ namespace graphene { namespace db {
                   catch ( const fc::exception& e )
                   {
                      elog( "${e}", ("e",e.to_detail_string() ) );
-                     throw; // maybe crash..
+                     //throw; // maybe crash..
                   }
```

Comment:
i am not very sure if we can just remove the `throw` here or replace it with `FC_THROW` with message and remove `elog`.

Reply:
As the comment indicates, the program crash at this point is deliberate. The problem is that the internal database may be in an inconsistent state at this point (i. e. the current session was only partially undone). The node cannot reliably continue to operate, so commenting this out is definitely wrong.

Reply:
So, the node should exit immediately, but why not just use std::exit?

Reply:
there is a (slight) chance that this throw does not actually end the program, and it has to do with stack unwinding. 

See https://akrzemi1.wordpress.com/2011/09/21/destructors-that-throw/

More thought should be put into what should happen in this case.

Reply:
anything commented "Maybe crash..." is something worth looking into imho :D

Reply:
@cwyyprog does GCC 8 complain that should not throw in destructor?

BTW other changes in this PR are fine.

Reply:
I know GCC 7.3.0 gives a warning about throw within a destructor. That's the main reason I created issue #1246

Reply:
The GCC 8 says throw in destructor will terminate the program, and that is now what we wanted, but why not exit the program in a normal way instead of throwing an exception implicitly? can we replace the throw with std::exit()?

Reply:
we cannot exit, terminate in libraries, in libraries we can only return or throw

Reply:
look please #1050 why we cannot exit or terminate in libraries